src/main/java/com/_up/megastore/services/implementations/SizeService.java

```diff
@@ -46,14 +46,13 @@ public SizeResponse updateSize(UUID sizeId,UpdateSizeRequest updateSizeRequest){
 
     @Override
     public SizeResponse restoreSize(UUID sizeId){
-        validateSizeForRestoration(sizeId);
         Size size = findSizeByIdOrThrowException(sizeId);
+        validateSizeForRestoration(size, sizeId);
         size.setDeleted(false);
         return SizeMapper.toSizeResponse(sizeRepository.save(size));
     }
 
-    public void validateSizeForRestoration(UUID sizeId){
-        Size size = findSizeByIdOrThrowException(sizeId);
+    public void validateSizeForRestoration(Size size, UUID sizeId){
         if(!size.isDeleted()){
             throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Size with id " + sizeId + " is not deleted." );
         }
```

Comment:
No hace falta pasar el sizeId si ya la podés obtener del size usando el getter
```suggestion
    public void validateSizeForRestoration(Size size){
        if(!size.isDeleted()){
            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Size with id " + size.getSizeId() + " is not deleted." );
        }
```